hive-event-listeners/apiary-gluesync-listener/src/main/java/com/expediagroup/apiary/extensions/gluesync/listener/ApiaryGluePreEventListener.java

```diff
@@ -22,34 +22,70 @@
+
 /**
  * A MetaStorePreEventListener that prevents actions not compatible with glue catalog.
  */
 
 public class ApiaryGluePreEventListener extends MetaStorePreEventListener {
 
   private static final Logger log = LoggerFactory.getLogger(ApiaryGluePreEventListener.class);
+  private final IsIcebergTablePredicate isIcebergPredicate = new IsIcebergTablePredicate();
+  private final String enableHiveToGlueRenameOperation;
 
   public ApiaryGluePreEventListener(Configuration config) {
     super(config);
+    enableHiveToGlueRenameOperation = System.getenv("ENABLE_HIVE_TO_GLUE_RENAME_OPERATION");
     log.debug("ApiaryGluePreEventListener created");
   }
 
   @Override
   public void onEvent(PreEventContext context) throws InvalidOperationException {
-    if (Objects.requireNonNull(context.getEventType()) != PreEventContext.PreEventType.ALTER_PARTITION) {
+    if (Objects.requireNonNull(context.getEventType()) != PreEventContext.PreEventType.ALTER_PARTITION &&
+        Objects.requireNonNull(context.getEventType()) != PreEventContext.PreEventType.ALTER_TABLE) {
       return;
     }
-
-    PreAlterPartitionEvent partitionEvent = (PreAlterPartitionEvent) context;
-    List<String> oldPartValues = partitionEvent.getOldPartVals();
-    List<String> newPartValues = partitionEvent.getNewPartition().getValues();
-    if (oldPartValues != null && !(newPartValues.equals(oldPartValues))) {
-      throw new InvalidOperationException("Rename Partition is not allowed when glue sync is enabled");
+    switch (context.getEventType()) {
+    case ALTER_TABLE:
+      PreAlterTableEvent tableEvent = (PreAlterTableEvent) context;
+      // Glue support iceberg renames, if Hive, check if rename workaround is enabled
+      if (enableHiveToGlueRenameOperation() || isIceberg(tableEvent)) {
+        break;
+      }
+      if (isRenameOperation(tableEvent)) {
+        throw new InvalidOperationException("Rename Table is not allowed when glue sync is enabled");
+      }
+      break;
+    case ALTER_PARTITION:
+      PreAlterPartitionEvent partitionEvent = (PreAlterPartitionEvent) context;
+      List<String> oldPartValues = partitionEvent.getOldPartVals();
+      List<String> newPartValues = partitionEvent.getNewPartition().getValues();
+      if (oldPartValues != null && !(newPartValues.equals(oldPartValues))) {
+        throw new InvalidOperationException("Rename Partition is not allowed when glue sync is enabled");
+      }
+      break;
+    default:
+      break;
     }
   }
+
+  private boolean isRenameOperation(PreAlterTableEvent tableEvent) {
+    return !(tableEvent.getOldTable().getTableName().equals(tableEvent.getNewTable().getTableName()));
+  }
+
+  private boolean isIceberg(PreAlterTableEvent tableEvent) {
+    return isIcebergPredicate.test(tableEvent.getOldTable().getParameters());
+  }
+
+  private boolean enableHiveToGlueRenameOperation() {
+    return enableHiveToGlueRenameOperation != null &&
```

Comment:
-> `return Boolean.valueOf(enableHiveToGlueRenameOperation)`

Reply:
Done for one part, but that does not work for "1" check

Reply:
ok remove "1" check? Why are we complicating this? Isn't one value easier?